Homework2/Program.cs

```diff
@@ -10,24 +10,72 @@ class Program
     {
         static void Main(string[] args)
         {
+            int nominal = 0;
+            double priceOrigin = 0;
+            double priceTrade = 0;
+
             do
             {
-                Console.WriteLine(new string('-', 30));
-                Console.WriteLine("Enter number of shares:");
-                int nominal = Convert.ToInt32(Console.ReadLine());
+                do
+                {
+                    try
+                    {
+                        Console.WriteLine(new string('-', 30));
+                        Console.WriteLine("Enter number of shares:");
+                        nominal = Convert.ToInt32(Console.ReadLine());
+                        if (nominal < 0)
+                        {
+                            throw new Exception("Number of shares must be bigger than 0. Try to enter number of shares one more time:");
+                        }
+                    }
+                    catch (Exception e)
+                    {
+                        Console.WriteLine(e.Message);
+                    }
+                } while (nominal < 0);
 
-                Console.WriteLine("Enter origin price per share:");
-                double priceOrigin = Convert.ToDouble(Console.ReadLine());
+                do
+                {
+                    try
+                    {
+                        Console.WriteLine(new string('-', 30));
+                        Console.WriteLine("Enter origin price per share:");
+                        priceOrigin = Convert.ToDouble(Console.ReadLine());
+                        if (priceOrigin < 0)
+                        {
+                            throw new Exception("Original price must be bigger than 0. Try to enter original price one more time:");
```

Comment:
The previous code with the placement of the throw keyword was correct.
The throw keyword must be in class CurrentValueCalc and try/catch must be in Program. If you can throw and catch in the same file it means that you can rewrite code without exception at all. And code without exceptions is preferred. If you write code for other developers with help of exceptions you can warn them on which way they should use your class. So developing code to give this code to another developer you will use throw meaning that other developers will use catch and will decide how to show an error to the end-user. Also, throw can be used to send messages between layers of application. From business logic to the presentation layer.